src/System.Drawing.Design.Primitives/src/System/Drawing/Design/ToolboxItem.cs

```diff
@@ -0,0 +1,1100 @@
+        {
+            if (this == obj)
+            {
+                return true;
+            }
+
+            if (obj == null)
+            {
+                return false;
+            }
+
+            if (!(obj.GetType() == this.GetType()))
+            {
+                return false;
+            }
+
+            ToolboxItem otherItem = (ToolboxItem)obj;
+
+            return TypeName == otherItem.TypeName &&
+                   AreAssemblyNamesEqual(AssemblyName, otherItem.AssemblyName) &&
+                   DisplayName == otherItem.DisplayName;
+        }
+
+
+        public override int GetHashCode()
+        {
+            string typeName = TypeName;
+            int hash = (typeName != null) ? typeName.GetHashCode() : 0;
+
+            return unchecked(hash ^ DisplayName.GetHashCode());
+        }
+
+        /// <summary>
+        /// Filters a property value before returning it.  This allows a property to always clone values,
+        ///    or to provide a default value when none exists.
+        /// </summary>
+        protected virtual object FilterPropertyValue(string propertyName, object value)
+        {
+            switch (propertyName)
+            {
+                case "AssemblyName":
+                    if (value != null) value = ((AssemblyName)value).Clone();
+
+                    break;
+
+                case "DisplayName":
+                case "TypeName":
+                    if (value == null) value = string.Empty;
+
+                    break;
+
+                case "Filter":
+                    if (value == null) value = new ToolboxItemFilterAttribute[0];
+
+                    break;
+
+                case "IsTransient":
+                    if (value == null) value = false;
+
+                    break;
+            }
+            return value;
+        }
+
+        /// <summary>
+        /// Allows access to the type associated with the toolbox item.
+        /// The designer host is used to access an implementation of ITypeResolutionService.
+        /// However, the loaded type is not added to the list of references in the designer host.
+        /// </summary>
+        public Type GetType(IDesignerHost host)
+        {
+            return GetType(host, AssemblyName, TypeName, false);
+        }
+
+        /// <summary>
+        /// This utility function can be used to load a type given a name.  AssemblyName and
+        ///     designer host can be null, but if they are present they will be used to help
+        ///     locate the type.  If reference is true, the given assembly name will be added
+        ///     to the designer host's set of references.
+        /// </summary>
+        protected virtual Type GetType(IDesignerHost host, AssemblyName assemblyName, string typeName, bool reference)
+        {
+            ITypeResolutionService ts = null;
+            Type type = null;
+
+            if (typeName == null)
+            {
+                throw new ArgumentNullException("typeName");
```

Comment:
Use nameof(typeName)

Reply:
Please use nameof, we do this in all the ports, it is more consistent to use nameof(variable) than the string itself.